include/deal.II/multigrid/mg_base.h

```diff
@@ -189,7 +189,22 @@ class MGTransferBase : public Subscriptor
   virtual void
   prolongate(const unsigned int to_level,
              VectorType &       dst,
-             const VectorType & src) const = 0;
+             const VectorType & src) const;
+
+  /**
+   * Prolongate a vector from level <tt>to_level-1</tt> to level
+   * <tt>to_level</tt>, summing into the previous content of <tt>dst</tt>.
+   *
+   * @arg src is a vector with as many elements as there are degrees of
+   * freedom on the coarser level involved.
+   *
+   * @arg dst has as many elements as there are degrees of freedom on the
+   * finer level.
+   */
+  virtual void
+  prolongate_and_add(const unsigned int to_level,
+                     VectorType &       dst,
+                     const VectorType & src) const;
```

Comment:
The other thing I would like to bring to discussion now is if we want to call this function `prolongate_and_add` (consistent with `restrict_and_add`) or `prolongate_add` (consistent with `vmult_add`). I guess we will go with the consistency within the multigrid transfer classes.

Reply:
I called it this way because of `restrict_and_add` (a function next to it). But, in your original code it was indeed `prolongate_add`. But, I don't have a strong opinion here.

Reply:
I agree, let us go with `prolongate_and_add`.